addons/web_tour/static/src/tour_service/tour_automatic.js

```diff
@@ -70,6 +70,11 @@ export class TourAutomatic {
                             );
                         },
                         action: async () => {
+                            try {
+                                await step.checkForUndeterminisms();
+                            } catch (error) {
+                                this.throwError(`UNDETERMINISM: ${error.message}`);
+                            }
```

Comment:
I would maybe do the whole logic in checkForUndeterminisms, to have here something like
```js
if (flag) { // if we introduce a flag to enable the check
    await step.checkForUndeterminisms();
}
```